Helper/Cart.php

```diff
@@ -1284,22 +1286,115 @@ public function getCartData($paymentOnly, $placeOrderPayload, $immutableQuote =
             $taxAmount = 0;
         }
 
+        $this->setLastImmutableQuote($immutableQuote);
+
         // include potential rounding difference and reset $diff accumulator
         $cart['items'][0]['total_amount'] += round($diff);
         $totalAmount += round($diff);
         $diff = 0;
 
         // add discount data
-        $cart['discounts'] = [];
+        list ($cart, $totalAmount, $diff) = $this->collectDiscounts (
+            $cart, $totalAmount, $diff, $immutableQuote, $address, $totals, $paymentOnly
+        );
 
+        /////////////////////////////////////////////////////////////////////////////////
+        // Add fixed amount type to all discounts if total amount is negative
+        // and set total to 0. Otherwise add calculated diff to cart total.
+        /////////////////////////////////////////////////////////////////////////////////
+        if ($totalAmount < 0) {
+            $totalAmount = 0;
+            foreach ($cart['discounts'] as &$discount) {
+                $discount['type'] = 'fixed_amount';
+            }
+        } else {
+            // add the diff to first item total to pass bolt order create check
+            $cart['items'][0]['total_amount'] += round($diff);
+            $totalAmount += round($diff);
+        }
+        /////////////////////////////////////////////////////////////////////////////////
+
+        $cart['total_amount'] = $totalAmount;
+        $cart['tax_amount']   = $taxAmount;
+
+        if (abs($diff) >= $this->treshold) {
+            $this->bugsnag->registerCallback(function ($report) use ($diff, $cart) {
+                $report->setMetaData([
+                    'TOTALS_DIFF' => [
+                        'diff' => $diff,
+                        'cart' => $cart,
+                    ]
+                ]);
+            });
+            $this->bugsnag->notifyError('Cart Totals Mismatch', "Totals adjusted by $diff.");
+        }
+
+        $this->sessionHelper->cacheFormKey($immutableQuote);
+
+        // $this->logHelper->addInfoLog(json_encode($cart, JSON_PRETTY_PRINT|JSON_UNESCAPED_SLASHES));
```

Comment:
Commented code here.

Reply:
@matt-thomason this is here for debugging purposes and always was. I turn it on / of during development. I will remove it if you say so.

Reply:
I don't love it but if it has a purpose then it's fine for this PR. We can perhaps revisit it when we're more generically cleaning up the plugin.